GVFS/GVFS.Virtualization/Projection/GitIndexProjection.GitIndexParser.cs

```diff
@@ -105,21 +134,50 @@ private FileSystemTaskResult AddToProjection(GitIndexEntry data)
                 return FileSystemTaskResult.Success;
             }
 
-            private FileSystemTaskResult AddToModifiedFiles(GitIndexEntry data)
+            /// <summary>
+            /// Adjusts the modifed paths and placeholders list for an index entry.
+            /// </summary>
+            /// <param name="gitIndexEntry">Index entry</param>
+            /// <param name="filePlaceholders">
+            /// Dictionary of file placeholders.  AdjustModifedPathsAndPlaceholdersForIndexEntry will
+            /// remove enties from filePlaceholders as they are found in the index.  After
+            /// AdjustModifedPathsAndPlaceholdersForIndexEntry is called for all entries in the index 
+            /// filePlaceholders will contain only those placeholders that are not in the index.
+            /// </param>
+            private FileSystemTaskResult AdjustModifedPathsAndPlaceholdersForIndexEntry(
+                GitIndexEntry gitIndexEntry,
+                Dictionary<string, PlaceholderListDatabase.PlaceholderData> filePlaceholders)
             {
-                if (!data.SkipWorktree)
+                gitIndexEntry.ParsePath();
+                string placeholderRelativePath = gitIndexEntry.GetRelativePath();
+
+                FileSystemTaskResult result = FileSystemTaskResult.Success;
+
+                if (!gitIndexEntry.SkipWorktree)
                 {
                     // A git command (e.g. 'git reset --mixed') may have cleared a file's skip worktree bit without
-                    // triggering an update to the projection.  Ensure this file is in GVFS's modified files database
-                    data.ParsePath();
-                    return this.projection.AddModifiedPath(data.GetFullPath());
+                    // triggering an update to the projection. If git cleared the skip-worktree bit then git will
+                    // be responsible for updating the file and we need to:
+                    //    - Ensure this file is in GVFS's modified files database
+                    //    - Remove this path from the placeholders list (if present)
+                    result = this.projection.AddModifiedPath(placeholderRelativePath);

```

Comment:
Ah yes missed that.  I just saw that this changed from using `GetFullPath` which used the git path separator to `GetRelativePath` which is using the OS path separator.  Do you think there will be any performance impact on Windows since it will need to replace separators now?  I don't think so since this seems to be a limited code path.

Reply:
> Do you think there will be any performance impact on Windows since it will need to replace separators now? I don't think so since this seems to be a limited code path.

You're right that this is a limited code path.  

We do have the option of getting both path formats, but then it's a trade off between the replacement and building a new path.

Your comment did remind me that I wanted to add some tracing to record how long this scan\update took.  I'll add that in.

If we find that in practice this new code is taking to long we can look for further optimizations, let me know if you think it's work doing more perf analysis now.